earth2studio/perturbation/hcbv.py

```diff
@@ -128,6 +132,10 @@ def create_generator(
             xper, _ = self.model(xper, coords)
 
             dx = xper - xunp
+
+            # remove zero elements, when inputs are constant
+            dx[..., dx.sum(dim=(0, 1, 2, 4, 5)) == 0, :, :] = torch.nan
```

Comment:
I don't know if I love having fixed integers here for the sum. Is there a way to use the coordinates to determine which dimensions to sum over?

Reply:
Ah great point, changed it 👍